File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            int ele=target-nums[i];
            if(mp.contains(ele)){
                return {mp[ele],i};
            }
            mp[nums[i]]=i;
        }
        return {-1,-1};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        HashMap<Integer,Integer> l=new HashMap<>();
        for(int i=0;i<n;i++){
            int rem=target-nums[i];
            if(l.containsKey(rem)){
                return new int[]{l.get(rem),i};
            }
            l.put(nums[i],i);
        }
        return new int[]{0,0};
    }
}